File: recursion/is-palindrome.js
function isPalindrome(str) {
  let newStr = str.split('').filter((letter) => letter.match(/[a-z]/i));
  let reverse = (arr) => {
    if(arr.length === 1) return arr[0];
    let newArr =  arr.slice(0, arr.length - 1)
    return arr[arr.length - 1] + reverse(newArr)
  }
  return reverse(newStr) == newStr.join('')
}

console.log(isPalindrome('heh! '))

File: recursion/is_palindrome.rb
def palindrome? (str)
  
  def reverse(str)
    if str.length == 1
      return str
    end

    last_letter = str[str.length - 1]
    new_string = str.split('').slice(0, str.length - 1).join('')

    last_letter + reverse(new_string)
  end
  

  str == reverse(str)  
end


p palindrome?('heh')